#!/usr/bin/env python
PACKAGE = "flypulator_traj_generator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k",     double_t, 0, "Drone: thrust constant",               0.0138, 0.000001,  0.2)
gen.add("b",     double_t, 0, "Drone: drag constant",               0.0000011, 0.000001,  0.2) 
gen.add("alpha",     double_t, 0, "Drone: tilting angle of rotors",               13.6, 0,  45) 
gen.add("beta",     double_t, 0, "Drone: tilting angle of rotors",               0, 0,  45) 
gen.add("length",     double_t, 0, "Drone: length of each rotor arm",               0.6, 0.1,  4)
gen.add("delta_h",     double_t, 0, "Drone: height between rotor and arm",               0, 0,  1)
gen.add("mass",     double_t, 0, "Drone: mass of the hexacopter",               4.398, 0.5,  10)
gen.add("i_xx",     double_t, 0, "Drone: Inertia w.r.t. x-Axis",               0.122, 0.1,  5)
gen.add("i_yy",     double_t, 0, "Drone: Inertia w.r.t. y-Axis",               0.123, 0.1,  5)
gen.add("i_zz",     double_t, 0, "Drone: Inertia w.r.t. z-Axis",               0.235, 0.1,  5)
gen.add("rotor_vel_max",     double_t, 0, "Drone: maximum rotor velocity",               5700, 100,  10000)
gen.add("rotor_vel_min",     double_t, 0, "Drone: minimum rotor velocity",               0, 0,  1000)

exit(gen.generate(PACKAGE, "flypulator_traj_generator", "traj_parameter"))
